packages/hierarchy-browser/src/hooks/useLoadEntitiesOnMountEffect.ts

```diff
@@ -13,10 +13,28 @@ export function useLoadEntitiesOnMountEffect(
 	) => void,
 	isOpen: boolean,
 	entitiesLoaded: boolean,
+	size?: number,
 ): void {
-	useLayoutEffect(() => {
-		if (isOpen && !entitiesLoaded) {
+	useEffect(() => {
+		if (isOpen && !entitiesLoaded && size && size > 0) {
```

Comment:
I don't think you need the `&& size` check. Undefined and null both return false for `> 0`